### PR TITLE
branch-AddGradedTest

### PR DESCRIPTION
**Pull request by raydenlim**
Added GradedTest.java
Added ReadingAssessment.java
Added MidTerms.java
Added Finals.java
Added PracticalExams.java
Added ExpectedGrade.java
Added JsonAdapetedGradedTest.java
Updated Persons.java
Updated JsonAdaptedPersonTest.java

### CODE REVIEW

**Comment by raydenlim** [comment]
Added GradedTestListBook.java
Added ReadOnlyGradedTestList.java
Added GradedTestList.java
Added GradedTEstNotFoundException.java
Added GradedTestListStorage.java
Added JsonGradedTestListStorage.java
Added JsonSerializableGradedTestList.java
Added GradedTestBuilder.java
Added TypicalGradedTest.java

Updated GradedTest.java
Updated ReadingAssessment.java
Updated MidTerms.java
Updated Finals.java
Updated PracticalExams.java
Updated ExpectedGrade.java
Updated JsonAdapetedGradedTest.java
Updated Persons.java
Updated JsonAdaptedPersonTest.java

**Comment by raydenlim** [comment]
Fixed * imports error
Added essential JavaDocs

File: Test Results - EditCommandParserTest_parse_allFieldsSpecified_success.html
Comment (Respirayson) [inline Test Results - EditCommandParserTest_parse_allFieldsSpecified_success.html:1]: why got this file

File: img.png
Comment (Respirayson) [inline img.png:1]: perhaps delete unused files

File: src/main/java/seedu/address/logic/parser/EditCommandParser.java [hunk 1]
```diff
@@ -82,4 +88,21 @@ private Optional<Set<Tag>> parseTagsForEdit(Collection<String> tags) throws Pars
...
     }
 
+    /**
+     * Parses {@code Collection<String> gradedTest} into a {@code Set<GradedTest>}
+     * if {@code gradedTestIndv} is non-empty.
+     * If {@code gradedTestIndv} contain only one element which is an empty string, it will be parsed into a
+     * {@code Set<GradedTest>} containing zero gradedTestIndv.
+     */
+    private Optional<Set<GradedTest>> parseGradedTestForEdit(Collection<String> gradedTestIndv) throws ParseException {
+        assert gradedTestIndv != null;
```
Comment (Respirayson) [inline src/main/java/seedu/address/logic/parser/EditCommandParser.java:98]: perhaps you want to use requireNonNull

**Review by Respirayson — APPROVED** [review]
LGTM!

**Pull request merged**